data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessor.java

```diff
@@ -12,22 +12,60 @@
 import com.amazon.dataprepper.model.processor.AbstractProcessor;
 import com.amazon.dataprepper.model.processor.Processor;
 import com.amazon.dataprepper.model.record.Record;
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 
 /**
  * Processor to parse CSV data in Events.
  *
  */
 @DataPrepperPlugin(name="csv", pluginType = Processor.class, pluginConfigurationType = CSVProcessorConfig.class)
 public class CSVProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private final CSVProcessorConfig config;
+
     @DataPrepperPluginConstructor
     public CSVProcessor(final PluginMetrics pluginMetrics, final CSVProcessorConfig config) {
         super(pluginMetrics);
-        throw new UnsupportedOperationException();
+        this.config = config;
     }
 
     @Override
     public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        final CsvMapper mapper = createCsvMapper();
+        final CsvSchema schema = createCsvSchema();
+
+        for (final Record<Event> record : records) {
+
+            final Event event = record.getData();
+
+            final String message = event.get(config.getSource(), String.class);
+            final boolean userDidSpecifyHeaderEventKey = (config.getColumnNamesSourceKey() != null);
```

Comment:
I think you could use `Objects.nonnull()` instead of `!= null` to be more declarative and help with readability

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#nonNull-java.lang.Object-